### ISSUE TITLE
Potential Data Leakage in Fine-Tuned LLM Evaluations?

### ISSUE THREAD

**Issue by slee-lab** [RyotaroOKabe/llm4syn#1]
Thank you for sharing [your work on arXiv](https://arxiv.org/html/2410.20976v1) and making the code publicly available. I would like to bring to your attention a potential concern regarding data leakage that may be leading to overestimated prediction accuracies across the fine-tuned LLM tasks.

TGT2CEQ task is particularly interesting due to its potential to recommend promising precursors for a given target. However, it is known that this is not a task with one clear answer (ground truth), because there can be multiple reaction pathways to synthesize a target material.

This led to me to take a closer look ([code](https://github.com/slee-lab/llm4syn/blob/main/investigate_data.ipynb)). Please note that I tried to reproduce your training and test set using the same configuration, including `random.seed(42)`. As a result, I observed that a significant portion of the test set labels overlap with the training set. This overlap suggests that the model may be memorizing rather than learning materials synthesis. To illustrate, for the case of CaCu3Ti4O12 (CCTO) —one of the examples highlighted in Table 3—I found the following:

1. The training set contains 77 instances of targeted synthesis for CCTO, indicating a substantial data leakage issue.
2. Furthermore, these 77 data points in the training set are dominated by the CuO + TiO2 + CaCO3 precursor set (i.e. 3 CuO + 4 TiO2 + 1 CaCO3 -> 1 CaCu3Ti4O12 + 1 CO2, , which I will refer to as **_CEQ-1_** for brevity), with 54/77 hits. This set also has 2 data points of 3 CuO + 4 TiO2 + 1 Ca(OH)2 -> 1 CaCu3Ti4O12 + 1 H2O (**_CEQ-2_**), which would be addressed below.
3. The test set consists of 10 data points of CEQ-1, and 1 data point of CEQ-2. This suggests that even within the test set, the "ground truth" is not technically CEQ-1 (Table 3), but instead reflects two possible CEQs.
4. Given points 2 and 3, it is likely that the fine-tuned GPT-2 model is simply learning to reproduce the most frequent completion (CEQ-1 in this case) from the training data rather than making insightful predictions. It may also be crucial to further investigate whether "indirect" data leakage is occurring, where similar synthesis targets lead to unintended memorization.

A similar trend appears across all the five other tasks, with test set and training set overlaps ranging from 6% to 33% depending on the task.

Given these findings, I believe it would be valuable to assess the extent of data leakage across all tasks and consider adjusting the evaluation methodology to ensure that model performance reflects true predictive capabilities.

I appreciate the effort that has gone into this work and look forward to your thoughts on this matter.

Thank you.
